telegram/ext/basepersistence.py

```diff
@@ -347,34 +352,75 @@ def update_conversation(
         """
 
     @abstractmethod
-    def update_user_data(self, user_id: int, data: Dict) -> None:
+    def update_user_data(self, user_id: int, data: UD) -> None:
         """Will be called by the :class:`telegram.ext.Dispatcher` after a handler has
         handled an update.
 
         Args:
             user_id (:obj:`int`): The user the data might have been changed for.
-            data (:obj:`dict`): The :attr:`telegram.ext.dispatcher.user_data` [user_id].
+            data (:class:`telegram.ext.utils.types.UD`): The
+                :attr:`telegram.ext.dispatcher.user_data` ``[user_id]``.
         """
 
     @abstractmethod
-    def update_chat_data(self, chat_id: int, data: Dict) -> None:
+    def update_chat_data(self, chat_id: int, data: CD) -> None:
         """Will be called by the :class:`telegram.ext.Dispatcher` after a handler has
         handled an update.
 
         Args:
             chat_id (:obj:`int`): The chat the data might have been changed for.
-            data (:obj:`dict`): The :attr:`telegram.ext.dispatcher.chat_data` [chat_id].
+            data (:class:`telegram.ext.utils.types.CD`): The
+                :attr:`telegram.ext.dispatcher.chat_data` ``[chat_id]``.
         """
 
     @abstractmethod
-    def update_bot_data(self, data: Dict) -> None:
+    def update_bot_data(self, data: BD) -> None:
         """Will be called by the :class:`telegram.ext.Dispatcher` after a handler has
         handled an update.
 
         Args:
-            data (:obj:`dict`): The :attr:`telegram.ext.dispatcher.bot_data` .
+            data (:class:`telegram.ext.utils.types.BD`): The
+                :attr:`telegram.ext.dispatcher.bot_data`.
         """
 
+    def refresh_user_data(self, user_id: int, user_data: UD) -> None:
+        """Will be called by the :class:`telegram.ext.Dispatcher` before passing the
+        :attr:`user_data` to a callback. Can be used to update data stored in :attr:`user_data`
+        from an external source.
+
+        Args:
+            user_id (:obj:`int`): The user ID this :attr:`user_data` is associated with.
+            user_data (:class:`telegram.ext.utils.types.UD`): The ``user_data`` of a single user.
+        """
+        raise NotImplementedError(
```

Comment:
jup, b/c that would be breaking: subclasses of ABC-subclasses must implement *all* abstract methods - otherwise they can't be instantiated. I'll add it to the todo list in the first comment that we add that to the deprecation issues (similar to #2231)

Reply:
ohh I see, I forgot about that